### PR TITLE
Add a hint for `Requires-Python`

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

As requested in the originating PR.

### CODE REVIEW

File: crates/uv/tests/lock.rs [hunk 1]
```diff
@@ -1004,6 +1004,8 @@ fn lock_requires_python() -> Result<()> {
            cannot be used, we can conclude that pygls>=1.1.0 cannot be used.
           And because project==0.1.0 depends on pygls>=1.1.0, we can conclude that project==0.1.0 cannot be used.
           And because only project==0.1.0 is available and project depends on project, we can conclude that the requirements are unsatisfiable.
+
+          hint: The `Requires-Python` requirement (>=3.7) defined in your `pyproject.toml` must be narrower than the `Requires-Python` of the resolved dependencies, but pygls>=1.1.0,<=1.2.1 only supports: >=3.7.9, <4
```
Comment (zanieb) [inline crates/uv/tests/lock.rs:1008]: This reads a little confusing to me, what does this mean? I must use `requires-python >=3.7.9`? Do I need the upper bound too? Perhaps "narrower" isn't intuitive enough for a user-facing message.

Comment (charliermarsh) [reply crates/uv/tests/lock.rs:1008]: The simplest explanation is: `>=3.7` includes versions that aren't included in `>=3.7.9, <4`. You have to use a narrower range, or change the pygls version. What do you suggest?

Comment (zanieb) [reply crates/uv/tests/lock.rs:1008]: Yeah that makes sense. I wonder if you should just say something more like that...

> The `Requires-Python` (>=3.7) defined in your `pyproject.toml` includes Python versions that are not supported by the resolved dependencies: pygls>=1.1.0,<=1.2.1 only supports Python >=3.7.9, <4. Use a narrower `Requires-Python` range or try a different `pygls` version.

Comment (zanieb) [reply crates/uv/tests/lock.rs:1008]: Sorry I don't have any ideas I love here. I was earnestly confused by the current message and imagine users that don't understand constraint ranges will be very lost.

Comment (charliermarsh) [reply crates/uv/tests/lock.rs:1008]: Totally fair. I will play with it.

Comment (charliermarsh) [reply crates/uv/tests/lock.rs]: It will only be displayed once (the fields are ignored when testing uniqueness).

Comment (charliermarsh) [reply crates/uv/tests/lock.rs]: > Separately... I worry about the upper bound, i.e. Poetry bounds to < 4 by default — does that mean if you have a single package published with Poetry in your dependency tree that you are forced to have this upper bound in your project?

Yeah, basically. I don't think there's anything we can do about this.

Comment (charliermarsh) [reply crates/uv/tests/lock.rs]: I guess we could special-case things a bit, like... if you have an open-ended `Requires-Python` requirement, treat it as compatible with `< 4`. What do you think?

Comment (charliermarsh) [reply crates/uv/tests/lock.rs]: Or, alternatively, throw out any `< 4` on `Requires-Python`.

Comment (charliermarsh) [reply crates/uv/tests/lock.rs]: Tracking here: https://github.com/astral-sh/uv/issues/4022

**Pull request merged**